lib/src/features/user/model/user.dart

```diff
@@ -211,86 +211,77 @@ class UserPerfStats with _$UserPerfStats {
+        maxPlayStreak: playStreak('nb', 'max').letOrNull(UserStreak.fromPick),
+        curTimeStreak: playStreak('time', 'cur').letOrNull(UserStreak.fromPick),
+        maxTimeStreak: playStreak('time', 'max').letOrNull(UserStreak.fromPick),
         worstLosses: stat('worstLosses', 'results')
             .asListOrNull((pick) => UserPerfGame.fromPick(pick)),
         bestWins: stat('bestWins', 'results')
             .asListOrNull((pick) => UserPerfGame.fromPick(pick)));
   }
 }
 
-abstract class UserStreak {
-  late final UserPerfGame? startGame;
-  late final UserPerfGame? endGame;
-
-  UserStreak(this.startGame, this.endGame);
-
-  UserStreak.fromPick(RequiredPick pick) {
-    startGame = pick('from').letOrNull(UserPerfGame.fromPick);
-    endGame = pick('to').letOrNull(UserPerfGame.fromPick);
+@freezed
+class UserStreak with _$UserStreak {
+  const factory UserStreak.gameStreak({
+    required int gamesPlayed,
+    required bool isValueEmpty,
+    required UserPerfGame? startGame,
+    required UserPerfGame? endGame,
+  }) = UserGameStreak;
+
+  const factory UserStreak.timeStreak({
+    required Duration timePlayed,
+    required bool isValueEmpty,
+    required UserPerfGame? startGame,
+    required UserPerfGame? endGame,
+  }) = UserTimeStreak;
+
+  factory UserStreak.fromJson(Map<String, dynamic> json) {
+    return UserStreak.fromPick(pick(json).required());
   }
 
-  bool isValueZero();
-}
-
-class UserGameStreak extends UserStreak {
-  final int gamesPlayed;
-
-  UserGameStreak(
-      {required this.gamesPlayed,
-      UserPerfGame? startGame,
-      UserPerfGame? endGame})
-      : super(startGame, endGame);
-
-  UserGameStreak.withPick(super.pick, {required this.gamesPlayed})
-      : super.fromPick();
-
-  @override
-  bool isValueZero() => gamesPlayed == 0;
-
-  factory UserGameStreak.fromJson(Map<String, dynamic> json) =>
-      UserGameStreak.fromPick(pick(json).required());
-
-  factory UserGameStreak.fromPick(RequiredPick pick) =>
-      UserGameStreak.withPick(pick, gamesPlayed: pick('v').asIntOrThrow());
-}
-
-class UserTimeStreak extends UserStreak {
-  final Duration timePlayed;
-
-  UserTimeStreak(
-      {required this.timePlayed,
-      UserPerfGame? startGame,
-      UserPerfGame? endGame})
-      : super(startGame, endGame);
-
-  UserTimeStreak.withPick(super.pick, {required this.timePlayed})
-      : super.fromPick();
-
-  @override
-  bool isValueZero() => timePlayed.inSeconds == 0;
-
-  factory UserTimeStreak.fromJson(Map<String, dynamic> json) =>
-      UserTimeStreak.fromPick(pick(json).required());
-
-  factory UserTimeStreak.fromPick(RequiredPick pick) =>
-      UserTimeStreak.withPick(pick,
-          timePlayed: pick('v').asDurationFromSecondsOrThrow());
+  factory UserStreak.fromPick(RequiredPick pick) {
+    final path = pick.path;
+    if (path.length <= 1) {
+      throw ArgumentError('pick must have a path longer than one');
```

Comment:
Here it is JSON decoding, so `ArgumentError` is not appropriate. Moreover in dart `Error` is not meant to be catched.
Cf: https://stackoverflow.com/questions/17315945/error-vs-exception-in-dart

It should be a `DataFormatException` or a `PickException` and the message should be more useful, like `'cannot decode $object as 'UserStreak''`